controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -60,27 +63,32 @@
 
             @Override
             public void connectionDropped() {
+                log.warn("CreateSegment {}/{}/{} Connection dropped", scope, stream, segmentNumber);
                 result.completeExceptionally(
                         new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
             }
 
             @Override
             public void wrongHost(WireCommands.WrongHost wrongHost) {
+                log.warn("CreateSegment {}/{}/{} wrong host", scope, stream, segmentNumber);
                 result.completeExceptionally(new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost));
             }
 
             @Override
             public void segmentAlreadyExists(WireCommands.SegmentAlreadyExists segmentAlreadyExists) {
+                log.info("CreateSegment {}/{}/{} segmentAlreadyExists", scope, stream, segmentNumber);
```

Comment:
I'm wondering if this is WARN rather than INFO. If this is never an issue, then INFO is fine, and otherwise WARN would be more appropriate.

Reply:
its idempotent, hence info.